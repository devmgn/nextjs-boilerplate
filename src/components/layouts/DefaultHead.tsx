import Head from 'next/head';
import appSettings from 'appSettings';

const DefaultHead = () => {
  return (
    <Head>
      <meta charSet="utf-8" />
      <meta name="viewport" content="width=device-width,initial-scale=1.0,minimum-scale=1.0" />
      <title>{appSettings.meta.appName}</title>
      <meta name="description" content={appSettings.meta.defaultDescription} />
      <link rel="icon" href="/images/favicon.ico" />
    </Head>
  );
};

export default DefaultHead;
