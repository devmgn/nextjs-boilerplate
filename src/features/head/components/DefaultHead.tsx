import { APP_NAME, META } from '@/config/appSettings';

const DefaultHead: React.FC = () => {
  return (
    <>
      <meta charSet="utf-8" />
      <meta name="viewport" content="width=device-width,initial-scale=1.0,minimum-scale=1.0" />
      <title>{APP_NAME}</title>
      <meta name="description" content={META.DEFAULT_DESCRIPTION} />
      <link rel="icon" href="/images/favicon.ico" />
    </>
  );
};

export default DefaultHead;
