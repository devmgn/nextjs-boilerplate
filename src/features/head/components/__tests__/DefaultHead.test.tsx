import { render } from '@testing-library/react';
import DefaultHead from '../DefaultHead';

describe('DefaultHead', () => {
  test('スナップショットテスト', () => {
    const { asFragment } = render(<DefaultHead />);
    expect(asFragment()).toMatchSnapshot();
  });
});
