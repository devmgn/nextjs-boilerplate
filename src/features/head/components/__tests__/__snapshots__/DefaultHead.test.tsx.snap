// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultHead スナップショットテスト 1`] = `
<DocumentFragment>
  <meta
    charset="utf-8"
  />
  <meta
    content="width=device-width,initial-scale=1.0,minimum-scale=1.0"
    name="viewport"
  />
  <title>
    app name [TEST]
  </title>
  <meta
    content="default description [TEST]"
    name="description"
  />
  <link
    href="/images/favicon.ico"
    rel="icon"
  />
</DocumentFragment>
`;
